src/components/WrapEtherBtn.tsx

```diff
@@ -349,6 +329,11 @@ const WrapUnwrapEtherBtn: React.FC<WrapUnwrapEtherBtnProps> = (props: WrapUnwrap
     ],
   })
 
+  // toggleModal recreated every time, keep ref to always use current in async code
```

Comment:
So it was that?? 
🤔but with the useState should have worked then too, right? 
event if the toggle is reconstructed, the hook `isModalVisible` should have the same value. Obviously my reasoning is failing somewhere

Reply:
`isModalVisible` is used inside `txOptionalParams` which is called asyncronously
So the `isModalVisible` inside `txOptionalParams` is actually  the one that is created at the same time as `txOptionalParams` function is created. `txOptionalParams` acts as closure around `isModalVisible` from that particular render.
And when `txOptionalParams` is finally called the real ``isModalVisible`` has been changed in a different render

1. Render 1:
1.1. `isModalVisible = A` created 
1.2. `txOptionalParams` and encloses over `isModalVisible = A` 
2. Render 2,3,4:
2.1.  `isModalVisible = B,C,D` created 
3. Render X:
3.1.   `txOptionalParams` called, still has ref to `isModalVisible = A`

Reply:
I see, thanks!